app/assets/javascripts/govuk_publishing_components/vendor/lux/lux.js

```diff
@@ -0,0 +1,835 @@
+        return !0
+    }
+
+    function _getCookie(e) {
+        try {
+            for (var t = document.cookie.split(";"), n = 0; n < t.length; n++) {
+                var r = t[n].split("=");
+                if (e === r[0].trim()) return unescape(r[1])
+            }
+        } catch (e) {
+            dlog("Error accessing document.cookie.")
+        }
+    }
+
+    function _setCookie(e, t, n) {
+        try {
+            document.cookie = e + "=" + escape(t) + (n ? "; max-age=" + n : "") + "; path=/; SameSite=Lax"
+        } catch (e) {
+            dlog("Error setting document.cookie.")
+        }
+    }
+
+    function _padLeft(e, t) {
+        return (t + e).slice(-t.length)
+    }
+
+    function dlog(e) {
+        gaLog.push(e), LUX.debug && console.log("LUX: " + e)
+    }
+    gaEventTypes.forEach((function(e) {
+        window.addEventListener(e, onInput, ghListenerOptions)
+    })), _auto && ("complete" == document.readyState ? _sendLux() : addListener("load", (function() {
+        setTimeout(_sendLux, 200)
+    })), addListener("beforeunload", _sendLux), addListener("unload", _sendLux), addListener("beforeunload", _sendIx), addListener("unload", _sendIx)), _addIxHandlers();
+    var _LUX = {
+        mark: _mark,
+        measure: _measure,
+        init: _init,
+        send: _sendLux,
+        addData: _addData,
+        getSessionId: _getUniqueId,
+        getDebug: function() {
+            return gaLog
+        },
+        forceSample: function() {
+            setUniqueId(createSyncId(!0)), console.log("Sampling has been turned on for this session.")
+        },
+        doUpdate: _doUpdate,
+        cmd: function(e) {
+            var t = e.shift();
+            "function" == typeof _LUX[t] && _LUX[t].apply(_LUX, e)
+        },
+        beaconUrl: _beaconUrl,
+        samplerate: _samplerate,
+        auto: _auto,
+        label: void 0 !== LUX.label ? LUX.label : void 0,
+        jspagelabel: void 0 !== LUX.jspagelabel ? LUX.jspagelabel : void 0,
+        version: version,
+        ae: [],
+        al: [],
+        debug: !!LUX.debug
+    };
+    return LUX.ac && LUX.ac.length && LUX.ac.forEach((function(e) {
+        var t = e.shift();
+        "function" == typeof _LUX[t] && _LUX[t].apply(_LUX, e)
+    })), void 0 !== window.LUX_ae && window.LUX_ae.forEach((function(e) {
+        errorHandler(e)
+    })), dlog("lux.js evaluation end."), _LUX
+}();
+var LUX_t_end = Date.now();
+
+// This ID usually appended to the end of the lux.js as a query string when
+// using the SpeedCurve hosted version - but we have to include it here as this
+// is self hosted.
+LUX.customerid = 47044334;
+
+// Turn on the image-based beacon, rather than have a remote JavaScript file
+// fetched and executed.
+LUX.beaconMode = "simple";
+
+// Setting debug to `true` shows what happening as it happens. Running
+// `LUX.getDebug()` in the browser's console will show the history of what's
+// happened.
+LUX.debug = false;
```

Comment:
This adds quite a lot of JS. Do we know exactly what the file size impact of this change will be?

Reply:
So the file sizes are:

| File | `rum-loader` | `lux-polyfill` | `lux` |
| - | - | - | - |
| Uncompressed | 1.5kb | 3.5kb | 33.5kb |
| Minified | 1.1kb | 2.3kb | 21.8kb | 

So the total minified is 25.2kb (and could be smaller when gzipped, but I can't find a way of turning that on locally).

All of which is loaded using the `async` attribute - so it won't block any loading or rendering of the page.

Reply:
Good numbers, and good to see you've got the stats. Worth adding to the docs?